src/function/table/read_csv.cpp

```diff
@@ -326,6 +326,25 @@ static unique_ptr<FunctionData> CSVReaderDeserialize(Deserializer &deserializer,
 	return std::move(result);
 }
 
+bool PushdownTypeToCSVScanner(LogicalGet &logical_get, const vector<LogicalType> &target_types,
+                              const vector<unique_ptr<Expression>> &expressions) {
+	auto &csv_bind = logical_get.bind_data->Cast<ReadCSVData>();
+	// We loop in the expressions and bail out if there is anything weird (i.e., not a bound column ref)
+	vector<LogicalType> pushdown_types = csv_bind.csv_types;
+	for (idx_t i = 0; i < expressions.size(); i++) {
+		if (expressions[i]->type == ExpressionType::BOUND_COLUMN_REF) {
```

Comment:
I think this doesn't handle the case where the same expression is referenced multiple times with different types, e.g.:

###### timestamp.csv
```
ts
2020-01-01 01:02:03
```

````sql
create table timestamps(ts timestamp, dt date);
insert into timestamps select ts,ts from read_csv('timestamp.csv');
from timestamps;
┌─────────────────────┬────────────┐
│         ts          │     dt     │
│      timestamp      │    date    │
├─────────────────────┼────────────┤
│ 2020-01-01 01:02:03 │ 2020-01-01 │
└─────────────────────┴────────────┘
```

Could we add this test?